apps/mymedicare_cb/views.py

```diff
@@ -56,63 +59,91 @@ def authenticate(request):
-
-    try:
-        sls_client.exchange(code, request)
-    except requests.exceptions.HTTPError as e:
-        logger.error("Token request response error {reason}".format(reason=e))
-        # Log also for info
-        log_authenticate_start(auth_flow_dict, "FAIL",
-                               "Token request response error {reason}".format(reason=e))
-        raise BBMyMedicareCallbackAuthenticateSlsClientException(ERROR_MSG_MYMEDICARE)
-
-    userinfo_endpoint = getattr(
-        settings,
-        'SLS_USERINFO_ENDPOINT',
-        'https://test.accounts.cms.gov/v1/oauth/userinfo')
-
-    headers = sls_client.auth_header()
-    # keep using deprecated conv - no conflict issue
-    headers.update({"X-SLS-starttime": str(datetime.datetime.utcnow())})
-    if request is not None:
-        headers.update({"X-Request-ID": str(getattr(request, '_logging_uuid', None)
-                        if hasattr(request, '_logging_uuid') else '')})
-
-    try:
-        response = requests.get(userinfo_endpoint,
-                                headers=headers,
-                                verify=sls_client.verify_ssl,
-                                hooks={
-                                    'response': [
-                                        response_hook_wrapper(sender=SLSUserInfoResponse,
-                                                              auth_flow_dict=auth_flow_dict)]})
-        response.raise_for_status()
-    except requests.exceptions.HTTPError as e:
-        logger.error("User info request response error {reason}".format(reason=e))
-        # Log also for info
-        log_authenticate_start(auth_flow_dict, "FAIL",
-                               "User info request response error {reason}".format(reason=e))
-        raise BBMyMedicareCallbackAuthenticateSlsClientException(ERROR_MSG_MYMEDICARE)
-
-    # Get the userinfo response object
-    user_info = response.json()
-
-    # Set identity values from userinfo response.
-    sls_subject = user_info.get("sub", "").strip()
-    sls_hicn = user_info.get("hicn", "").strip()
-    #     Convert SLS's mbi to UPPER case.
-    sls_mbi = user_info.get("mbi", "").strip().upper()
-    sls_first_name = user_info.get('given_name', "")
-    sls_last_name = user_info.get('family_name', "")
-    sls_email = user_info.get('email', "")
+    # SLS vs. SLSx flow based on feature switch slsx-enable (true = SLSx / false = SLS)
+    if waffle.switch_is_active('slsx-enable'):
+        request_token = request.GET.get('req_token', None)
+        if request_token is None:
+            log_authenticate_start(auth_flow_dict, "FAIL",
+                                   "SLSx request_token is missing in callback error.")
+            raise ValidationError(settings.MEDICARE_ERROR_MSG)
+        slsx_client = OAuth2ConfigSLSx()
+        try:
+            access_token, user_id = slsx_client.exchange_for_access_token(request_token, request)
+        except requests.exceptions.HTTPError as e:
```

Comment:
Similar to the other HTTPError exception, does this do something raise_for_status doesnt?  Perhaps just wrap it in our own custom exception?  Just checking.

Reply:
Could this be moved inside the slsx_client for consolidation with the other exceptions (unlike the SLS client which doesnt do any exception checking there), or is there a more logical reason it should be here?